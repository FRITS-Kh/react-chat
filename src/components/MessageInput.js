import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Input from '@material-ui/core/Input';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';

const styles = theme => ({
  messageInputWrap: {
    position: 'fixed',
    right: theme.spacing.unit * 3,
    bottom: theme.spacing.unit * 3,
    left: 344,
    padding: theme.spacing.unit * 2,
  },
  messageInput: {
    width: '100%',
  },
});

class MessageInput extends React.Component {
  static propTypes = {
    sendMessage: PropTypes.func.isRequired,
    classes: PropTypes.objectOf(PropTypes.string).isRequired,
    showJoinButton: PropTypes.bool.isRequired,
    onJoinButtonClick: PropTypes.func.isRequired,
    disabled: PropTypes.bool.isRequired,
  };

  state = {
    value: '',
  };

  handleValueChange = (event) => {
    this.setState({
      value: event.target.value,
    });
  };

  handleKeyPress = (event) => {
    const { value } = this.state;

    if (event.key === 'Enter' && value) {
      this.props.sendMessage(value);
      this.setState({ value: '' });
    }
  };

  render() {
    const {
      classes, showJoinButton, onJoinButtonClick, disabled,
    } = this.props;

    return (
      <Paper className={classes.messageInputWrap} elevation={6}>
        {showJoinButton ? (
          <Button
            disabled={disabled}
            fullWidth
            variant="raised"
            color="primary"
            onClick={onJoinButtonClick}
          >
            Join
          </Button>
        ) : (
          <Input
            disabled={disabled}
            placeholder="Type your messageâ€¦"
            className={classes.messageInput}
            inputProps={{
              'aria-label': 'Description',
            }}
            value={this.state.value}
            onChange={this.handleValueChange}
            onKeyPress={this.handleKeyPress}
          />
        )}
      </Paper>
    );
  }
}

export default withStyles(styles)(MessageInput);
