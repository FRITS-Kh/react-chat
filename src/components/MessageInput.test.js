/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MessageInput from './MessageInput';

const mockProps = {
  showJoinButton: false,
  disabled: false,
  onJoinButtonClick: jest.fn(),
  sendMessage: jest.fn(),
};

describe('<MessageInput />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageInput {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders input', () => {
    const tree = renderer.create(<MessageInput {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders disabled input', () => {
    const tree = renderer.create(<MessageInput {...mockProps} disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders button', () => {
    const tree = renderer.create(<MessageInput {...mockProps} showJoinButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders disabled button', () => {
    const tree = renderer.create(<MessageInput {...mockProps} showJoinButton disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
