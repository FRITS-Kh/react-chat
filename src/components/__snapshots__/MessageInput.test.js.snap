// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageInput /> renders button 1`] = `
<div
  className="MuiPaper-root-45 MuiPaper-elevation6-53 MuiPaper-rounded-46 MessageInput-messageInputWrap-43"
>
  <button
    className="MuiButtonBase-root-107 MuiButton-root-85 MuiButton-contained-94 MuiButton-containedPrimary-95 MuiButton-raised-97 MuiButton-raisedPrimary-98 MuiButton-fullWidth-106"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-86"
    >
      Join
    </span>
    <span
      className="MuiTouchRipple-root-110"
    />
  </button>
</div>
`;

exports[`<MessageInput /> renders disabled button 1`] = `
<div
  className="MuiPaper-root-45 MuiPaper-elevation6-53 MuiPaper-rounded-46 MessageInput-messageInputWrap-43"
>
  <button
    className="MuiButtonBase-root-107 MuiButtonBase-disabled-108 MuiButton-root-85 MuiButton-contained-94 MuiButton-containedPrimary-95 MuiButton-raised-97 MuiButton-raisedPrimary-98 MuiButton-disabled-101 MuiButton-fullWidth-106"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="-1"
    type="button"
  >
    <span
      className="MuiButton-label-86"
    >
      Join
    </span>
  </button>
</div>
`;

exports[`<MessageInput /> renders disabled input 1`] = `
<div
  className="MuiPaper-root-45 MuiPaper-elevation6-53 MuiPaper-rounded-46 MessageInput-messageInputWrap-43"
>
  <div
    className="MuiInput-root-72 MuiInput-disabled-75 MuiInput-underline-76 MessageInput-messageInput-44"
    onKeyPress={[Function]}
  >
    <input
      aria-invalid={undefined}
      aria-label="Description"
      aria-required={undefined}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInput-input-80 MuiInput-disabled-75"
      defaultValue={undefined}
      disabled={true}
      id={undefined}
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder="Type your message…"
      readOnly={undefined}
      required={undefined}
      rows={undefined}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`<MessageInput /> renders input 1`] = `
<div
  className="MuiPaper-root-45 MuiPaper-elevation6-53 MuiPaper-rounded-46 MessageInput-messageInputWrap-43"
>
  <div
    className="MuiInput-root-72 MuiInput-underline-76 MessageInput-messageInput-44"
    onKeyPress={[Function]}
  >
    <input
      aria-invalid={undefined}
      aria-label="Description"
      aria-required={undefined}
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInput-input-80"
      defaultValue={undefined}
      disabled={false}
      id={undefined}
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder="Type your message…"
      readOnly={undefined}
      required={undefined}
      rows={undefined}
      type="text"
      value=""
    />
  </div>
</div>
`;
